libs/experimental/langchain_experimental/data_anonymizer/presidio.py

```diff
@@ -81,11 +122,193 @@ def _anonymize(self, text: str) -> str:
             operators=self.operators,
         ).text
 
-    def add_recognizer(self, recognizer: EntityRecognizer) -> None:
-        """Add a recognizer to the analyzer"""
-        self._analyzer.registry.add_recognizer(recognizer)
-        self.analyzed_fields.extend(recognizer.supported_entities)
 
-    def add_operators(self, operators: Dict[str, OperatorConfig]) -> None:
-        """Add operators to the anonymizer"""
-        self.operators.update(operators)
+class PresidioReversibleAnonymizer(PresidioAnonymizerBase, ReversibleAnonymizerBase):
+    def __init__(
+        self,
+        analyzed_fields: Optional[List[str]] = None,
+        language: str = "en",
+        operators: Optional[Dict[str, OperatorConfig]] = None,
+        faker_seed: Optional[int] = None,
+    ):
+        super().__init__(analyzed_fields, language, operators, faker_seed)
+        self._deanonymizer_mapping: Dict[str, Dict[str, str]] = defaultdict(
+            lambda: defaultdict(str)
+        )
+
+    @property
+    def deanonymizer_mapping(self) -> Dict[str, Dict[str, str]]:
+        """Return the deanonymizer mapping"""
+        return {k: dict(v) for k, v in self._deanonymizer_mapping.items()}
+
+    def _create_update_deanonymizer_mapping(
```

Comment:
i would say it **updates** the mapping as it is already created after instance initialization - it is just empty